### PR TITLE
Debian packaging

### PR DESCRIPTION
**Pull request by mangelajo**

### CODE REVIEW

**Comment by mangelajo** [comment]
PPA says:
```
Rejected:
flightctl_2024.02.13.dsc: Unknown section 'flightctl'
flightctl_2024.02.13.tar.gz: Unknown section 'flightctl'
flightctl_2024.02.13_source.buildinfo: Unknown section 'flightctl'
Further error processing not possible because of a critical previous error.

flightctl (2024.02.13) jammy; urgency=medium

  * Debian ppa/deb packaging.

===

If you don't understand why your files were rejected please send an email
to [launchpad-users@lists.launchpad.net](mailto:launchpad-users@lists.launchpad.net) for help (requires membership).
```

**Comment by mangelajo** [comment]
> PPA says:
> 
> ```
> Rejected:
> flightctl_2024.02.13.dsc: Unknown section 'flightctl'
> flightctl_2024.02.13.tar.gz: Unknown section 'flightctl'
> flightctl_2024.02.13_source.buildinfo: Unknown section 'flightctl'
> Further error processing not possible because of a critical previous error.
> 
> flightctl (2024.02.13) jammy; urgency=medium
> 
>   * Debian ppa/deb packaging.
> 
> ===
> 
> If you don't understand why your files were rejected please send an email
> to [launchpad-users@lists.launchpad.net](mailto:launchpad-users@lists.launchpad.net) for help (requires membership).
> ```

ok so Section is the category of the package, flightctl makes no sense there. I moved it to admin section.

File: .github/workflows/ppa-build.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+jobs:
+  build_and_upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y devscripts debhelper make golang-1.20-go gpg dh-golang
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply .github/workflows/ppa-build.yaml]: Oh, good catch. I thought it was not available, but it is!.

**Comment by mangelajo** [comment]
All good, I have removed sending to ppa on PRs, but this is the previous run:
https://github.com/flightctl/flightctl/actions/runs/7904854008/job/21576142147

We can do local building on PRs, I'd prefer to do it as a follow up.

**Comment by mangelajo** [comment]
here's the ppa: https://launchpad.net/~flightctl/+archive/ubuntu/flightctl/+packages

**Review by hexfusion — APPROVED** [review]
I am not an expert in debian package building but this looks sane.

File: Makefile [hunk 2]
```diff
@@ -64,14 +65,42 @@ rpm:
 	packit build locally
 	mv $(shell uname -m)/flightctl-*.rpm bin/rpm
 
+
+# cross-building for deb pkg
+bin/amd64:
+	GOARCH=amd64 go build -buildvcs=false $(GO_BUILD_FLAGS) -o $@/flightctl ./cmd/flightctl/...
+	GOARCH=amd64 go build -buildvcs=false $(GO_BUILD_FLAGS) -o $@/flightctl-agent ./cmd/flightctl-agent/...
```
Comment (hexfusion) [inline Makefile:72]: do we want to include the server or is that separate?

Comment (mangelajo) [reply Makefile:72]: The idea is that server would be provided as container, this is only to be installed in systems that we want to manage.

**Pull request merged**